File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/entities/enemies/units/SmolderingTitan.java
package it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.units;

import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.Skills;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.StatusEffect;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.StatusEffectType;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy;
import it.paranoidsquirrels.idleguildmaster.storage.data.items.ItemWrapper;
import java.util.LinkedHashMap;

/* JADX INFO: loaded from: classes3.dex */
public class SmolderingTitan extends Enemy {
    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected int getMaxDamage() {
        return 550;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected int getMinDamage() {
        return 500;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.Entity
    public boolean isMagic() {
        return false;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.Entity
    public boolean isRanged() {
        return false;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected void configureStatistics() {
        this.baseMaxHp = 6850;
        this.baseConstitution = 340;
        this.baseIntelligence = 28;
        this.baseDexterity = 10;
        this.baseDefense = 35;
        this.baseMagicDefense = 0;
        this.statusImmunities.add(StatusEffectType.ABLAZE);
        this.onSelfHit = new StatusEffect(StatusEffectType.ABLAZE, null, 1, 1.0d);
        this.imageId = R.drawable.unit_smoldering_titan;
        this.idName = R.string.enemy_smoldering_titan_name;
        this.idDescription = R.string.enemy_smoldering_titan_description;
        this.passiveSkill = Skills.PASSIVE_FIRE_GIANT;
        this.activeSkill = Skills.ACTIVE_NONE;
        this.rarity = 1;
        this.expGiven = 3750;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    public LinkedHashMap<ItemWrapper, Integer> listDrops(int i) {
        LinkedHashMap<ItemWrapper, Integer> linkedHashMap = new LinkedHashMap<>();
        linkedHashMap.put(ItemWrapper.getInstance("Infernite", 1), 1000);
        return linkedHashMap;
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/entities/enemies/units/SmolderingTitan.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.units

import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.Skills
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.StatusEffect
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.StatusEffectType
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
import it.paranoidsquirrels.idleguildmaster.storage.data.items.ItemWrapper
import java.util.LinkedHashMap

class SmolderingTitan : Enemy() {
    override fun getMaxDamage(): Int = 550
    override fun getMinDamage(): Int = 500
    override fun isMagic(): Boolean = false
    override fun isRanged(): Boolean = false

    override fun configureStatistics() {
        baseMaxHp = 6850
        baseConstitution = 340
        baseIntelligence = 28
        baseDexterity = 10
        baseDefense = 35
        baseMagicDefense = 0
        statusImmunities.add(StatusEffectType.ABLAZE)
        onSelfHit = StatusEffect(StatusEffectType.ABLAZE, null, 1, 1.0)
        imageId = R.drawable.unit_smoldering_titan
        idName = R.string.enemy_smoldering_titan_name
        idDescription = R.string.enemy_smoldering_titan_description
        passiveSkill = Skills.PASSIVE_FIRE_GIANT
        activeSkill = Skills.ACTIVE_NONE
        rarity = 1
        expGiven = 3750
    }

    override fun listDrops(i: Int): LinkedHashMap<ItemWrapper, Int> {
        val linkedHashMap = LinkedHashMap<ItemWrapper, Int>()
        linkedHashMap.put(ItemWrapper.getInstance("Infernite", 1), 1000)
        return linkedHashMap
    }
}
